stix2/properties.py

```diff
@@ -17,82 +23,109 @@ class Property(object):
-          ValueError if the value is not valid.
-          (Default: if `clean` is defined, it will attempt to call `clean` and
-          return the result or pass on a ValueError that `clean` raises. If
-          `clean` is not defined, this will return `value` unmodified).
+        - Return a value that is valid for this property. If `value` is not
+          valid for this property, this will attempt to transform it first. If
+          `value` is not valid and no such transformation is possible, it should
+          raise a ValueError.
     - `def default(self):`
         - provide a default value for this property.
         - `default()` can return the special value `NOW` to use the current
             time. This is useful when several timestamps in the same object need
             to use the same default value, so calling now() for each field--
             likely several microseconds apart-- does not work.
 
-    Subclasses can instead provide lambda functions for `clean`, and `default`
-    as keyword arguments. `validate` should not be provided as a lambda since
-    lambdas cannot raise their own exceptions.
+    Subclasses can instead provide a lambda function for `default as a keyword
+    argument. `clean` should not be provided as a lambda since lambdas cannot
+    raise their own exceptions.
     """
 
-    def _default_validate(self, value):
+    def _default_clean(self, value):
         if value != self._fixed_value:
             raise ValueError("must equal '{0}'.".format(self._fixed_value))
         return value
 
-    def __init__(self, required=False, fixed=None, clean=None, default=None, type=None):
+    def __init__(self, required=False, fixed=None, default=None, type=None):
         self.required = required
         self.type = type
         if fixed:
             self._fixed_value = fixed
-            self.validate = self._default_validate
+            self.clean = self._default_clean
             self.default = lambda: fixed
-        if clean:
-            self.clean = clean
         if default:
             self.default = default
 
     def clean(self, value):
-        raise NotImplementedError
-
-    def validate(self, value):
-        try:
-            value = self.clean(value)
-        except NotImplementedError:
-            pass
         return value
 
+    def __call__(self, value=None):
+        if value is not None:
+            return value
+
 
 class ListProperty(Property):
 
-    def __init__(self, contained, required=False, element_type=None):
+    def __init__(self, contained, **kwargs):
         """
-        contained should be a type whose constructor creates an object from the value
+        Contained should be a function which returns an object from the value.
         """
-        self.contained = contained
-        self.element_type = element_type
-        super(ListProperty, self).__init__(required)
+        if contained == StringProperty:
```

Comment:
The documentation for self.contained sounds exactly like it's describing Property.clean ("returns an object from the value"... kinda vague, but I think that's what it means).  Why wouldn't that be suitable for all properties?  Also, if contained/clean is responsible for coercing/converting to a particular type, what role does Property.\_\_call\_\_ play?  The method isn't documented, and I can't tell from the code.

Also, creating a throwaway object just so you can call an instance method is ugly.  It implies that if that method ever uses `self`, this code is probably wrong.  But if clean() is not allowed to use `self`, maybe it should be a static or class method.

Finally, can I create a list of lists?  (E.g. ListProperty whose "contained" is another ListProperty.)  The code doesn't check for it, and it looks like it may error out (if contained is ListProperty, contained() is constructing a ListProperty without the correct constructor arguments).  More generally, that code seems to assume that all properties have a no-arg constructor.  And that's not the case.